lib/pages/App.js

```diff
@@ -173,12 +213,43 @@ export default class App extends Component {
             targetId={uploadTargetUser}
             timezoneIsSelected={Boolean(this.props.selectedTimezone)}
             userDropdownShowing={this.props.showingUserSelectionDropdown}
+            isClinicAccount={this.props.isClinicAccount}
             userIsSelected={uploadTargetUser !== null}
             addDevice={this.props.sync.addTargetDevice}
             removeDevice={this.props.sync.removeTargetDevice}
             onDone={this.props.async.clickDeviceSelectionDone} />
         </div>
       );
+    } else if (page === pages.CLINIC_USER_SELECT) {
+      const { allUsers, page, targetUsersForUpload, uploadTargetUser } = this.props;
+      return (
+        <div>
+          <ClinicUserSelect
+            allUsers={allUsers}
+            onUserChange={this.props.async.setUploadTargetUserAndMaybeRedirect}
```

Comment:
I'm not sure reusing `setUploadTargetUserAndMaybeRedirect` is really the best approach here. We end up dispatching SET_UPLOAD_TARGET_USER twice and the redirect logic is more complex. Let's discuss.